pkg/actors/actors.go

```diff
@@ -917,8 +917,14 @@ func (a *actorsRuntime) startReminder(reminder *Reminder, stopChannel chan bool)
 			if repetitionsLeft > 0 {
 				repetitionsLeft--
 			}
-			if err = a.updateReminderTrack(actorKey, reminder.Name, repetitionsLeft, nextTime); err != nil {
-				log.Errorf("error updating reminder track: %v", err)
+			_, exists = a.activeReminders.Load(reminderKey)
```

Comment:
The issue would still occur just less often. We could put one lock around reminder updates and deletes but I am not sure of the performance effect since deletes would block the all reminder evaluations/updates for the actor. Maybe adding reminder specific locks could work but I would need to think about that more.

Reply:
I would at least put a lock before line 921. After you check if the reminder still exists, you can't be sure it will exist 1ms later.

Reply:
The update reminder track should use eTag matching instead. This is how we fix the race condition for state stores in Dapr runtime.